acceptance/ui/features/steps/services_steps.rb

```diff
@@ -1,11 +1,97 @@
-Then (/^VHost "(.*?)" should exist$/) do |vhost|
+Then (/^VHost "([^"]*)" should exist$/) do |vhost|
     expect(checkVHostExists(vhost))
 end
 
-Then (/^Public Port "(.*?)" should exist$/) do |port|
+Then (/^Public Port "([^"]*)" should exist$/) do |port|
     expect(checkPublicPortExists(port))
 end
 
+Then (/^I should see the Add Public Endpoint dialog$/) do
+    initServicePageObj()
+    expect(@service_page.addPublicEndpoint_dialog.visible?).to be true
+end
+
+Then (/^I choose Port type$/) do
+    @service_page.find("div.btn-group", "//input[@value='port']" ).click
```

Comment:
Can this be redefined so that we can do something like `@service_page.portTypeButton.click`?

As much as possible, I'm trying to keep the CSS/xpath expressions in the page object model definitions.